pkg/autoscaler/scaling/autoscaler.go

```diff
@@ -245,8 +244,49 @@ func (a *Autoscaler) Scale(ctx context.Context, now time.Time) (desiredPodCount
 	return desiredPodCount, int32(excessBCF), true
 }
 
+func clampInclusiveRange(valueToClamp float64, lowerBound float64, upperBound float64) float64 {
+	return math.Min(math.Max(valueToClamp, lowerBound), upperBound)
+}
+
+// Query functions
+func (a *Autoscaler) currentlyStable() bool {
```

Comment:
This feels arbitrary. Why did `isOverPanicThreshold` got singled out?
Though logic here is simpler.

Reply:
I'm not sure I follow this feedback. Could you elaborate?

Reply:
Computation is `isOverPanicThreshold` is still inlined, while this got pulled into a function. though, `IsOverPanicThreshold` is more complex in its nature

Reply:
Gotcha. I'll extract functions.

Reply:
Actually I'm not sure this would be an improvement. L198 in the PR right now:
```go
isOverPanicThreshold && a.currentlyStable()
```

Which pretty much expresses the idea that if you're over the threshold and stable, you need to panic.

When converted to a function:

```go
isOverPanicThreshold(observedPanicValue, readyPodsCount, spec) && a.currentlyStable()
```

I have to scan a lot further to connect the two predicates. The effect is similar for the other case. One alternative is to move the `isOverPanicThreshold()` method to the end position, but I think it should have primacy over the others.

Another alternative is to hoist all the boolean logic into two all-encompassing predicate methods, so that L198-202 would look like this:

```go
if a.shouldStartPanicking(observedPanicValue, readyPodsCount, spec) {
	a.startPanicking(now, logger)
} else if a.shouldStopPanicking(observedPanicValue, readyPodsCount, spec, now) {
	a.stopPanicking(logger)
}
```

I'm not a fan of this either. I think personally I'd wind up bouncing up and down the call stack to see what those values do. In particular, this approach obscures the primacy of whether or not the panic threshold has been exceeded.

I guess this is a long way of saying that I now prefer the slightly-odd mix of variables and methods/functions to fully extracting to methods/functions only.

Reply:
I actually was deviously advocating to remove new functions 
but 🤷‍♂  this is minor.,

Reply:
Github needs a :mindblown: emoji

Reply:
I think I had stable/panicking as methods because I was concerned about the values changing between invocations, but that's only a problem between each call to Scale(), right? I'll switch those to variables.

Reply:
Ah, now I remember. Those values _do_ change if a panic starts or ends, so a boolean value captured before that will be wrong if there's a change.